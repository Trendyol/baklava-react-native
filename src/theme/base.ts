export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
