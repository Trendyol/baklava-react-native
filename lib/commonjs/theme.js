"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.space = exports.radii = exports.fontSizes = exports.default = exports.colors = void 0;
const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];
exports.space = space;
const fontSizes = [12, 14, 16, 20, 24, 28, 32];
exports.fontSizes = fontSizes;
const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16
};
exports.radii = radii;
const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
  // Gradient
  gradientPurple: ['#8C4EFF', '#5794FF'],
  gradientOrange: ['#FFA964', '#F27A1A']
};
exports.colors = colors;
var _default = {
  space,
  fontSizes,
  radii,
  colors
};
exports.default = _default;
//# sourceMappingURL=theme.js.map