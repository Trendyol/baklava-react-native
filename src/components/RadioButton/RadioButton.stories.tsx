import { ComponentMeta, ComponentStory } from '@storybook/react-native';
import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import RadioButton from './RadioButton';

const RadioButtonMeta: ComponentMeta<typeof RadioButton> = {
  title: 'RadioButton',
  component: RadioButton,
  argTypes: {
    onPress: { action: 'pressed the button' },
  },
  args: {
    selected: true,
    disabled: false,
  },
};

export default RadioButtonMeta;

type RadioButtonStory = ComponentStory<typeof RadioButton>;

export const Basic: RadioButtonStory = args => {
  const [selected, setSelected] = React.useState(false);

  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        RadioButton
      </Text>
      <RadioButton
        {...args}
        onPress={() => setSelected(true)}
        selected={selected}
        label="lorem ipsum"
        m="2xs"
      />
    </>
  );
};

export const Cases: RadioButtonStory = () => {
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        RadioButton
      </Text>
      <Box px="m" py="2xs" flexDirection="row">
        <RadioButton selected={false} />
        <Box width={8} />
        <RadioButton selected={true} />
        <Box width={8} />
        <RadioButton disabled selected={false} />
        <Box width={8} />
        <RadioButton disabled selected={true} />
      </Box>

      <Box px="m" py="2xs">
        <Box height={12} />
        <RadioButton selected={false} label="selected: false" />
        <Box height={12} />
        <RadioButton selected={true} label="selected: true" />
      </Box>

      <Box px="m" py="xs">
        <Text py="xs" variant="subtitle02Medium">
          Disable:
        </Text>
        <RadioButton
          disabled={true}
          selected={false}
          label="disabled, selected: false"
        />
        <Box height={12} />
        <RadioButton
          disabled={true}
          selected={true}
          label="disabled, selected: true"
        />
      </Box>

      <Box px="m" py="xs">
        <Text py="xs" variant="subtitle02Medium">
          Alignment:
        </Text>
        <RadioButton
          selected={false}
          label="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        />
        <Box height={12} />
        <RadioButton
          selected={false}
          alignItems="center"
          label="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        />
      </Box>
    </>
  );
};

export const RadioButtonWithFlagIcon: RadioButtonStory = () => {
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Radio Button With Flag Icon
      </Text>

      <Box px="m" py="2xs">
        <Box height={12} />
        <RadioButton
          selected={false}
          label="selected: false, flagIcon: TR"
          flagIcon="TR"
        />
        <Box height={12} />
        <RadioButton
          selected={true}
          label="selected: true, flagIcon: DE"
          flagIcon="DE"
        />
        <Box height={12} />
        <RadioButton
          disabled
          selected={false}
          label="selected: false, disabled, flagIcon: GR"
          flagIcon="GR"
        />
        <Box height={12} />
        <RadioButton
          disabled
          selected={true}
          label="selected: true, disabled, flagIcon: RO"
          flagIcon="RO"
        />
      </Box>
    </>
  );
};
