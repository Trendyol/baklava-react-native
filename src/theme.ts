import { createTheme } from '@shopify/restyle';

const theme = createTheme({
  colors: {
    white: '#ffffff',
    black: '#000000',
    transparent: 'transparent',
    // Color
    primaryColor: '#F27A1A',
    secondaryColor: '#273142',
    tertiaryColor: '#F1F2F7',
    successColor: '#0BC15C',
    dangerColor: '#FF5043',
    warningColor: '#FFB600',
    alternativeColor: '#5794FF',
    featuredColor: '#8C4EFF',
    // Hover
    primaryHover: '#EF6114',
    secondaryHover: '#0F131A',
    tertiaryHover: '#D5D9E1',
    successHover: '#09A44E',
    dangerHover: '#FF3028',
    warningHover: '#FF9800',
    alternativeHover: '#457EFF',
    featuredHover: '#753EFF',
    // Background
    primaryBackground: '#F1F2F7',
    accentBackground: '#FEF2E8',
    secondaryBackground: '#FFFFFF',
    tertiaryBackground: '#F7F7FA',
    successBackground: '#E7F9EF',
    alertBackground: '#FFEEEC',
    warningBackground: '#FFF8E6',
    infoBackground: '#EEF4FF',
    featuredCBackground: '#F4EDFF',
    // Border Color
    borderColor: '#D5D9E1',
    // Content
    contentPrimary: '#273142',
    contentSecondary: '#6E7787',
    contentTertiary: '#95A1B5',
    contentPassive: '#AFBBCA',
  },
  spacing: {
    none: 0,
    '4xs': 2,
    '3xs': 4,
    '2xs': 8,
    xs: 12,
    s: 14,
    m: 16,
    l: 20,
    xl: 24,
    '2xl': 32,
    '3xl': 40,
    '4xl': 48,
    '5xl': 64,
    '6xl': 80,
  },
  borderRadii: {
    none: 0,
    xs: 2,
    s: 4,
    m: 6,
    l: 8,
    xl: 16,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  fonts: {
    light: 'System',
    regular: 'System',
    medium: 'System',
    semiBold: 'System',
    bold: 'System',
  },
  textVariants: {
    defaults: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      color: 'contentPrimary',
    },
    // Heading
    heading1: {
      fontSize: 32,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 36,
    },
    heading2: {
      fontSize: 28,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 32,
    },
    heading3: {
      fontSize: 24,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 28,
    },
    // Subtitle01
    subtitle01Regular: {
      fontSize: 20,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 24,
    },
    subtitle01Medium: {
      fontSize: 20,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 24,
    },
    subtitle01Semibold: {
      fontSize: 20,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 24,
    },
    subtitle01Bold: {
      fontSize: 20,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 24,
    },
    // Subtitle02
    subtitle02Regular: {
      fontSize: 16,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 20,
    },
    subtitle02Medium: {
      fontSize: 16,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 20,
    },
    subtitle02Semibold: {
      fontSize: 16,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 20,
    },
    subtitle02Bold: {
      fontSize: 16,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 20,
    },
    // Subtitle03
    subtitle03Regular: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    subtitle03Medium: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
    },
    subtitle03Semibold: {
      fontSize: 14,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 16,
    },
    subtitle03Bold: {
      fontSize: 14,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 16,
    },
    // Subtitle04
    subtitle04Regular: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 14,
    },
    subtitle04Medium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 14,
    },
    subtitle04Semibold: {
      fontSize: 12,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 14,
    },
    subtitle04Bold: {
      fontSize: 12,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 14,
    },
    // Body
    bodyText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    bodyUnderline: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyTextLink: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyLongText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    // Caption
    captionText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionMedium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
    },
    captionLongText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionTextLink: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
      textDecorationLine: 'underline',
    },
  },
});

export type Theme = typeof theme;
export default theme;
