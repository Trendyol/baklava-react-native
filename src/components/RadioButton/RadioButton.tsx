import React from 'react';
import {
  LayoutProps,
  SpacingProps,
  SpacingShorthandProps,
  VariantProps,
  createRestyleComponent,
  createVariant,
  layout,
  spacing,
  spacingShorthand,
} from '@ergenekonyigit/restyle';
import { TouchableOpacity } from 'react-native';
import { Theme } from '../../theme';
import Box from '../Box/Box';
import Text from '../Text/Text';
import Icon from '../Icon/Icon';
import { IconNameType } from '../Icon/types';

const variant = createVariant({ themeKey: 'radioButtonVariants' });

type SelectboxProps = VariantProps<Theme, 'radioButtonVariants'> &
  React.ComponentProps<typeof Box>;

const Selectbox = createRestyleComponent<SelectboxProps, Theme>([variant], Box);

type ContainerProps = React.ComponentProps<typeof TouchableOpacity> &
  LayoutProps<Theme> &
  SpacingProps<Theme> &
  SpacingShorthandProps<Theme>;

const RadioButtonContainer = createRestyleComponent<ContainerProps, Theme>(
  [layout, spacing, spacingShorthand],
  TouchableOpacity,
);

type RadioButtonProps = React.ComponentProps<typeof RadioButtonContainer> & {
  selected: boolean;
  label?: string;
  disabled?: boolean;
  icon?: IconNameType | null;
  onPress?: () => void;
};

const RadioButton = ({
  selected,
  label,
  disabled = false,
  icon,
  onPress,
  ...rest
}: RadioButtonProps) => {
  const selectedBox = React.useMemo(() => {
    if (!selected) {
      return null;
    }

    return (
      <Box
        testID="bullet"
        borderRadius="full"
        width={8}
        height={8}
        backgroundColor={disabled ? 'neutralLighter' : 'neutralFull'}
      />
    );
  }, [disabled, selected]);

  const selectVariant = React.useMemo(() => {
    if (selected) {
      return disabled ? 'disabledSelected' : 'selected';
    } else {
      return disabled ? 'disabledUnselected' : 'unselected';
    }
  }, [selected, disabled]);

  const selectLabelColor = React.useMemo(() => {
    if (disabled) {
      return 'neutralLighter';
    } else {
      return selected ? 'primaryKey' : 'neutralDarker';
    }
  }, [selected, disabled]);

  return (
    <RadioButtonContainer
      flexDirection="row"
      alignSelf="flex-start"
      disabled={disabled}
      onPress={onPress}
      {...rest}>
      <Selectbox testID="selectbox" variant={selectVariant}>
        {selectedBox}
      </Selectbox>
      {icon ? (
        <Box marginLeft="2xs">
          <Icon
            size="s"
            name={icon}
            color={
              disabled
                ? 'neutralLight'
                : selected
                ? 'primaryKey'
                : 'neutralDarker'
            }
            testID="icon"
          />
        </Box>
      ) : null}
      {label ? (
        <Text testID="label" marginLeft="2xs" color={selectLabelColor}>
          {label}
        </Text>
      ) : null}
    </RadioButtonContainer>
  );
};

export default RadioButton;
