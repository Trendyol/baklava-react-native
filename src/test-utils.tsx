import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
