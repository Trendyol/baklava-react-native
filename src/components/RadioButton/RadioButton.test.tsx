import React from 'react';
import theme from '../../theme';
import { fireEvent, render } from '../../test-utils';
import RadioButton from './RadioButton';

describe('RadioButton', () => {
  beforeEach(() => {});

  test('should render RadioButton correctly when selected is false', () => {
    // when
    const { toJSON, getByTestId } = render(<RadioButton selected={false} />);
    const radiobuttonComponent = getByTestId('selectbox');

    // then
    expect(toJSON()).toMatchSnapshot();
    expect(radiobuttonComponent.props.style[1][0].borderColor).toBe(
      theme.colors.neutralLighter,
    );
    expect(radiobuttonComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.neutralFull,
    );
  });

  test('should render RadioButton correctly when selected is true', () => {
    // when
    const { toJSON, getByTestId } = render(<RadioButton selected={true} />);
    const radiobuttonComponent = getByTestId('selectbox');

    // then
    expect(toJSON()).toMatchSnapshot();
    expect(radiobuttonComponent.props.style[1][0].borderColor).toBe(
      theme.colors.primaryKey,
    );
    expect(radiobuttonComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.primaryKey,
    );
  });

  test('should render RadioButton correctly when disable, selected is false', () => {
    // when
    const { toJSON } = render(<RadioButton disabled={true} selected={false} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render RadioButton correctly when disable, selected is true', () => {
    // when
    const { toJSON } = render(<RadioButton disabled selected={true} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should press RadioButton', () => {
    // given
    const onPress = jest.fn();

    // when
    const { getByTestId } = render(
      <RadioButton selected={true} testID="radioButton" onPress={onPress} />,
    );
    const radiobuttonComponent = getByTestId('radioButton');
    fireEvent.press(radiobuttonComponent);

    // then
    expect(onPress).toBeCalledTimes(1);
  });

  test('should not press radiobutton when disabled', () => {
    // given
    const onPress = jest.fn();

    // when
    const { getByTestId } = render(
      <RadioButton
        selected={true}
        disabled
        testID="radiobuttonbox"
        label="lorem ipsum"
        onPress={onPress}
      />,
    );
    const radiobuttonComponent = getByTestId('radiobuttonbox');
    const radiobuttonLabelComponent = getByTestId('label');
    fireEvent.press(radiobuttonComponent);

    // then
    expect(onPress).not.toBeCalled();
    expect(radiobuttonLabelComponent.props.style[0].color).toBe(
      theme.colors.neutralLighter,
    );
  });

  test('should render without selectbox correctly', () => {
    // when
    const { queryByTestId } = render(<RadioButton selected={false} />);
    const radiobuttonSelectbox = queryByTestId('selectbox');

    // then
    expect(radiobuttonSelectbox.props.style[1][0].backgroundColor).toBe(
      theme.colors.neutralFull,
    );
  });

  test('should render with label correctly when selected', () => {
    // when
    const { getByText } = render(
      <RadioButton selected={true} label="lorem ipsum" />,
    );
    const radiobuttonLabel = getByText('lorem ipsum');

    // then
    expect(radiobuttonLabel).toBeTruthy();
    expect(radiobuttonLabel.props.style[0].color).toBe(theme.colors.primaryKey);
  });

  test('should render with label correctly when unselected', () => {
    // when
    const { getByText } = render(
      <RadioButton selected={false} label="lorem ipsum" />,
    );
    const radiobuttonLabel = getByText('lorem ipsum');

    // then
    expect(radiobuttonLabel).toBeTruthy();
    expect(radiobuttonLabel.props.style[0].color).toBe(
      theme.colors.neutralDarker,
    );
  });

  test('should render without label correctly', () => {
    // when
    const { queryByTestId } = render(<RadioButton selected={true} />);
    const radioButtonLabel = queryByTestId('');

    // then
    expect(radioButtonLabel).toBeFalsy();
  });

  test('should render with icon correctly when selected', () => {
    // when
    const { getByTestId } = render(
      <RadioButton selected={true} flagIcon="TR" />,
    );
    const iconComponent = getByTestId('flag-icon');

    // then
    expect(iconComponent).toBeTruthy();
  });

  test('should render with icon correctly when unselected', () => {
    // when
    const { getByTestId } = render(
      <RadioButton selected={false} flagIcon="TR" />,
    );
    const iconComponent = getByTestId('flag-icon');

    // then
    expect(iconComponent).toBeTruthy();
  });

  test('should render with icon correctly when disabled', () => {
    // when
    const { getByTestId } = render(
      <RadioButton selected={false} disabled flagIcon="TR" />,
    );
    const iconComponent = getByTestId('flag-icon');

    // then
    expect(iconComponent).toBeTruthy();
  });

  test('should render without icon correctly', () => {
    // when
    const { queryByTestId } = render(<RadioButton selected={true} />);
    const iconComponent = queryByTestId('flag-icon');

    // then
    expect(iconComponent).toBeFalsy();
  });
});
