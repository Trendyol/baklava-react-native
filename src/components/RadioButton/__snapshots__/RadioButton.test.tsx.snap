// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render RadioButton correctly when disable, selected is false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="selectbox"
    accessible={true}
    style={
      [
        {},
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F1F2F7",
            "borderColor": "#AFBBCA",
            "borderRadius": 999,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="selectbox"
  />
</View>
`;

exports[`RadioButton should render RadioButton correctly when disable, selected is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="selectbox"
    accessible={true}
    style={
      [
        {},
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F1F2F7",
            "borderColor": "#AFBBCA",
            "borderRadius": 999,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="selectbox"
  >
    <View
      accessibilityLabel="bullet"
      accessible={true}
      style={
        [
          {
            "backgroundColor": "#AFBBCA",
            "borderRadius": 999,
            "height": 8,
            "width": 8,
          },
        ]
      }
      testID="bullet"
    />
  </View>
</View>
`;

exports[`RadioButton should render RadioButton correctly when selected is false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="selectbox"
    accessible={true}
    style={
      [
        {},
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#AFBBCA",
            "borderRadius": 999,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="selectbox"
  />
</View>
`;

exports[`RadioButton should render RadioButton correctly when selected is true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="selectbox"
    accessible={true}
    style={
      [
        {},
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F27A1A",
            "borderColor": "#F27A1A",
            "borderRadius": 999,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="selectbox"
  >
    <View
      accessibilityLabel="bullet"
      accessible={true}
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 999,
            "height": 8,
            "width": 8,
          },
        ]
      }
      testID="bullet"
    />
  </View>
</View>
`;
